File: scripts/start-bot.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Start Bot Polling System Continuously

Wrapper script to run bot-poll in a loop.
Use this to keep squad agents processing tasks.

Usage: python scripts/start-bot.py
"""

import subprocess
import time
import signal
import sys
import os

# Set UTF-8 encoding for Windows console
if sys.platform == 'win32':
    os.system('chcp 65001 >nul 2>&1')

POLL_INTERVAL_SECONDS = 120  # 2 minutes

print("\n" + "=" * 60)
print("  Starting Squad Bot Polling System")
print("=" * 60)
print(f"  Polling every {POLL_INTERVAL_SECONDS} seconds...")
print("  Press Ctrl+C to stop.\n")

poll_count = 0

def run_poll():
    global poll_count
    poll_count += 1

    timestamp = time.strftime("%H:%M:%S")
    print(f"\n{'-' * 60}")
    print(f"  Poll #{poll_count} | {timestamp}")
    print('-' * 60)

    try:
        result = subprocess.run(
            ['npm', 'run', 'bot-poll'],
            capture_output=True,
            text=True,
            shell=True,
            encoding='utf-8',
            errors='ignore'
        )

        if result.returncode != 0:
            print(f"\n  Poll #{poll_count} failed with code {result.returncode}")
            if result.stderr:
                print(f"  Error: {result.stderr}")
    except Exception as e:
        print(f"\n  Error running poll: {e}")

def signal_handler(sig, frame):
    print(f"\n\nStopping bot polling system...")
    print(f"  Total polls completed: {poll_count}")
    print("  Goodbye!")
    print("\n" + "=" * 60)
    sys.exit(0)

# Register signal handlers
signal.signal(signal.SIGINT, signal_handler)
signal.signal(signal.SIGTERM, signal_handler)

# Run first poll immediately
run_poll()

# Set up interval for continuous polling
try:
    while True:
        time.sleep(POLL_INTERVAL_SECONDS)
        run_poll()
except KeyboardInterrupt:
    signal_handler(signal.SIGINT, None)

File: scripts/start-bot.js
/**
 * Start Bot Polling System Continuously
 *
 * Wrapper script to run bot-poll in a loop.
 * Use this to keep the squad agents processing tasks.
 *
 * Usage: node scripts/start-bot.js
 */

const { spawn } = require('child_process');

const POLL_INTERVAL_SECONDS = 120; // 2 minutes

console.log('\n🚀 Starting Squad Bot Polling System...');
console.log(`⏰ Polling every ${POLL_INTERVAL_SECONDS} seconds...\n`);
console.log('Press Ctrl+C to stop.\n');

let pollCount = 0;

function runPoll() {
  pollCount++;

  const timestamp = new Date().toISOString();
  const timeStr = new Date().toLocaleTimeString();

  console.log(`\n${'─'.repeat(60)}`);
  console.log(`  🔄 Poll #${pollCount} | ${timeStr}`);
  console.log('─'.repeat(60));

  const child = spawn('npm', ['run', 'bot-poll'], {
    stdio: 'inherit',
    shell: true,
  });

  child.on('close', (code) => {
    if (code !== 0) {
      console.error(`\n❌ Poll #${pollCount} failed with code ${code}`);
    }
  });
}

// Run first poll immediately
runPoll();

// Set up interval for continuous polling
const intervalId = setInterval(() => {
  runPoll();
}, POLL_INTERVAL_SECONDS * 1000);

// Handle graceful shutdown
process.on('SIGINT', () => {
  console.log('\n\n🛑 Stopping bot polling system...');
  clearInterval(intervalId);
  console.log(`\n✅ Total polls completed: ${pollCount}`);
  console.log('Goodbye! 👋\n');
  process.exit(0);
});

process.on('SIGTERM', () => {
  console.log('\n\n🛑 Stopping bot polling system...');
  clearInterval(intervalId);
  console.log(`\n✅ Total polls completed: ${pollCount}`);
  console.log('Goodbye! 👋\n');
  process.exit(0);
});
